akka-actor/src/main/scala-2.13-/akka/util/internal/SameThreadExecutionContext.scala

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.util.internal
+
+import akka.actor.ReflectiveDynamicAccess
+import akka.annotation.InternalApi
+
+import scala.concurrent.ExecutionContext
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] object SameThreadExecutionContext {
+  def apply(): ExecutionContext = {
```

Comment:
Do you think there is a risk that we start using `SameThreadExecutionContext()` in more places instead of `ExecutionContexts.sameThreadExecutionContext` and then this reflection stuff will run each time? Worth adding a comment at least, or put it in a `private val`.

Reply:
Not so worried but I'll detailing that in a bit of Scaladoc.

Reply:
Added docs, moved into `akka.dispatch.internal` and made it `private[dispatch]`